File: Clases2023-24/Clase26_ViewPager2Tabs_Fragments/app/src/main/java/com/example/clase26_viewpager2tabs_fragments/MyAdaptadorViewpager2Fragments.java
package com.example.clase26_viewpager2tabs_fragments;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class MyAdaptadorViewpager2Fragments extends FragmentStateAdapter {

    public MyAdaptadorViewpager2Fragments(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return new FragmentUno();
            case 1:
                return new FragmentDos();
            case 2:
                return new FragmentTres();
            case 3:
                return new FragmentCuatro();
        }

        return null;
    }

    @Override
    public int getItemCount() {
        return 4;
    }
}

File: Clases2024-25/Clase21_Viewpager2_/app/src/main/java/com/example/clase21_viewpager2_/MyAdaptadorViewpager2Fragments.kt
package com.example.clase21_viewpager2_

import androidx.viewpager2.adapter.FragmentStateAdapter

class MyAdaptadorViewpager2Fragments(mainActivity: MainActivity) : FragmentStateAdapter(mainActivity) {

    override fun createFragment(position: Int) = when (position) {
        0 -> FragmentUno()
        1 -> FragmentDos()
        2 -> FragmentTres()
        3 -> FragmentCuatro()
        else -> FragmentUno() // Default case
    }

    override fun getItemCount(): Int = 4

}
